examples/simple-example/README.md

```diff
@@ -35,3 +35,66 @@ miranum-core. The project consists of three modules:*
 Now you can simply start the process instance and view the log in the application console.
 
 <p align="right">(<a href="#top">back to top</a>)</p>
+
+### Generate element templates
+(*note: this feature is currently only available for Camunda 7.*)
+
+This example also demonstrates how you can make use of the miranum element-templates generation feature
+which lets you generate element templates for your worker definitions. \
+To use the element template generation you have to add the @GenerateElementTemplates annotation to the method of your Miranum worker.
+
+To be able to use this annotation you have to add the miranum-core dependency or just the element-template-api dependency to your maven project.
+
+```xml
+<dependency>
+   <groupId>io.miragon.miranum</groupId>
+   <artifactId>element-template-api</artifactId>
+   <version>0.1.0-SNAPSHOT</version>
+</dependency>
+```
+
+Some properties are required when using the @GenerateElementTemplate annotation:
+
+```java
+@Worker(type = "sendMessage")
+@GenerateElementTemplate(name = "Send Message",
+        id = "send-message",
+        type = "sendMessage",
+        appliesTo = {BPMNElementType.BPMN_SERVICE_TASK},
+        version = "0.0.1")
+public Answer sendMessage(final SendMessageCommand message) {
+        return new Answer("answer to: " + message.getName());
+}
+```
+
+Additionally, when deciding for which Workflow-Engine you are going to build your project for you have to add the concrete
+dependency of the element-templates generator to your maven project. \
+Below is the maven dependency needed for Camunda Platform 7::
+
+```xml
+<dependency>
+   <groupId>io.miragon.miranum</groupId>
+   <artifactId>element-templates-c7</artifactId>
+   <version>0.1.0-SNAPSHOT</version>
+</dependency>
+```
+
+To further customize the generated element templates you can add the @ElementTemplateProperty annotation to the input and output parameter of your worker definition. \
+For example we can change the name, type and constraints on the properties of the SendMessageCommand:
+
+```java
+class SendMessageCommand {
+    @ElementTemplateProperty(name = "Message", type = "String", required = true)
+    private String name;
+
+    @ElementTemplateProperty(name = "Correlation Key", type = "String", required = true)
+    private String key;
+}
+```
+
+The element templates are then generated on startup of the main project. \
+Currently they are stored in the `classes/element-templates` directory inside the `target` folder of the main project.
```

Comment:
```suggestion
Currently they are stored in the `classes/element-templates` directory inside the `target` folder of the main project. It might be necessary to reload the directory from disk to be able to see the generated element-template when using IntelliJ. 
```